daemon/cmd/daemon_main.go

```diff
@@ -1143,6 +1143,9 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 	flags.StringSlice(option.NodeLabels, []string{}, "List of label prefixes used to determine identity of a node (used only when enable-node-selector-labels is enabled)")
 	option.BindEnv(vp, option.NodeLabels)
 
+	flags.Int(option.EnableIPOptionTracing, 0, "Enable packet IP tracing. Set this option to specify the IP options field of IPv4 packet from which to read tracing information; a value of 0 disables IP tracing.")
```

Comment:
`GetUint` and you don't need to specify negative numbers are unsupported.

and the config.go usage below cannot be silently ignored.

Reply:
In config.go we only set the macro if the `<type>` set is positive because we use this as the macro to do parsing if it is defined.

If we change the config such that it always defines the macro with the value passed in by the user, this means that we would have to change how we do the parsing:
```
#ifdef ENABLE_PACKET_IP_TRACING
	check_and_store_ip_trace_id(ctx, ENABLE_PACKET_IP_TRACING);
#endif
```

We can do that, but this change just removes the check from the configuration into the bpf code instead, which I don't see a lot of value in.

Reply:
As a user, what if I set `-136`? Current implementation ignores that intent and treats it as `0`. Changing this to use `Uint` would cause the agent to error out and report that an invalid type is used for the option.

Further, this should be `Uint8` to constrain the input to 0-255 (the IPv4 option type range)

https://github.com/spf13/pflag/blob/d5e0c0615acee7028e1e2740a11102313be88de1/uint8.go#L81

Reply:
Agreed. Code has been changed to reflect this.